str/helper/merge_str_vcf_combiner.py

```diff
@@ -0,0 +1,168 @@
+        22,
+        24,
+        25,
+        26,
+        27,
+        28,
+        29,
+        30,
+        31,
+        32,
+        33,
+        35,
+        36,
+        37,
+        38,
+        39,
+        40,
+        41,
+        42,
+        43,
+        44,
+        46,
+    ]
+
+    # Initialize variables to store information
+    fileformat_line = ''
+    info_lines = []
+    alt_lines = set()
+    chrom_line = ''
+    gt_lines = []
+
+    shard_counter = 0
+    # Process each input file
+    for key in sorted_key_order:
+        input_file = to_path(input_files_dict[key])
+        print(f'Parsing {input_file}')
+        shard_counter = shard_counter + 1
+
+        with gzip.open(input_file, 'rt') as f:
+            for line in f:
+                # Collect information from the header lines
+                if line.startswith('##fileformat'):
+                    if key == 1:  # first file processed is shard_1
+                        fileformat_line = line
+                elif (
+                    line.startswith('##INFO')
+                    or line.startswith('##FILTER')
+                    or line.startswith('##FORMAT')
+                    or line.startswith('##contig')
+                    or line.startswith('##command')
+                ):
+                    if key == 1:
+                        info_lines.append(line)
+                elif line.startswith('##ALT'):
+                    # Collect ALT lines from all files into a set to remove duplicates
+                    alt_lines.add(line)
+                elif line.startswith('#CHROM'):
+                    if key == 1:
+                        chrom_line = line
+                elif not line.startswith('#'):
+                    # Collect calls after #CHROM
+                    gt_lines.append(line)
+
+    print(f'Parsed {shard_counter} sharded VCFs')
+    # Sort ALT lines alphabetically and convert to a list
+    sorted_alt_lines = sorted(alt_lines)
+
+    # Write the combined information to the output file
+    with to_path(gcs_out_path).open('w') as out_file:
+        # Write fileformat line
+        out_file.write(fileformat_line)
+        # Write INFO, FILTER, and FORMAT lines
+        out_file.writelines(info_lines)
+        # Write ALT lines
+        out_file.writelines(sorted_alt_lines)
+        # Write CHROM line
+        out_file.write(chrom_line)
+        # Write GT or lines containing the calls
+        out_file.writelines(gt_lines)
+
+
+@click.command()
+@click.option(
+    '--input-dir',
+    help='Parent input directory for sharded VCFs',
+)
+def main(input_dir):
+    """
+    Takes an input directory containing vcf shards
+    Aggregates all sharded data into a single output file
+    """
+
+    # Initializing Batch
+    b = get_batch()
+
+    combiner_job = b.new_python_job(name=f'VCF Combiner job')
+    out_path = output_path(f'combined_eh.vcf', 'analysis')
+    combiner_job.call(combine_vcf_files, input_dir, out_path)
+
+    b.run(wait=False)
```

Comment:
```suggestion
    out_path = output_path(f'combined_eh.vcf', 'analysis')
    combine_vcf_files(input_dir, out_path)
```

No need for this - we're already running a python task in a runtime which has all the auth required to get on with the job. Makes no sense for the whole purpose of this Python job to be generating a separate python job in a complex way